javagen/src/main/java/com/azure/autorest/template/StreamSerializationModelTemplate.java

```diff
@@ -1290,7 +1290,8 @@ private static void createObjectAndReturn(JavaBlock methodBlock, String modelNam
         if (propertiesManager.hasConstructorArguments()) {
             if (propertiesManager.getSetterPropertiesCount() == 0
                 && propertiesManager.getReadOnlyPropertiesCount() == 0
-                && propertiesManager.getAdditionalProperties() == null) {
+                && propertiesManager.getAdditionalProperties() == null
+                && propertiesManager.getSuperAdditionalPropertiesProperty() == null) {
```

Comment:
If the model as `additionalProperties` in superclass, it still need to set it separately.